src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/grid_sample/grid_sample_kernel_ref.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "grid_sample_kernel_ref.hpp"
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+namespace {
+
+CommonDispatchData SetDefault(const grid_sample_params& kernel_params) {
+    CommonDispatchData dispatch_data;
+    const auto& output = kernel_params.outputs.front();
+
+    dispatch_data.gws = {output.Batch().v * output.Feature().v, output.Y().v, output.X().v};
+    dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, kernel_params.engineInfo);
+
+    return dispatch_data;
+}
+
+}  // namespace
+
+KernelsData GridSampleKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+
+    auto kernel_data = KernelData::Default<grid_sample_params>(params);
+    const auto& kernel_params = dynamic_cast<const grid_sample_params&>(*kernel_data.params);
+    const auto dispatch_data = SetDefault(kernel_params);
+    const auto entry_point = GetEntryPoint(kernelName, kernel_params.layerID, params, options);
+    const auto jit_constants = GetJitConstants(kernel_params);
+    const auto jit = CreateJit(kernelName, jit_constants, entry_point);
+    auto& kernel = kernel_data.kernels.front();
+
+    FillCLKernelData(kernel, dispatch_data, params.engineInfo, kernelName, jit, entry_point, {}, false, false, 2);
+
+    return {kernel_data};
+}
+
+ParamsKey GridSampleKernelRef::GetSupportedKey() const {
+    ParamsKey key;
+    key.EnableAllInputDataType();
+    key.EnableAllOutputDataType();
+    key.EnableDifferentTypes();
+    key.EnableAllInputLayout();
+    key.EnableAllOutputLayout();
+    key.EnableTensorOffset();
+    key.EnableTensorPitches();
+    key.EnableBatching();
+    return key;
+}
+
+bool GridSampleKernelRef::Validate(const Params& params, const optional_params& options) const {
+    if (params.GetType() != KernelType::GRID_SAMPLE || options.GetType() != KernelType::GRID_SAMPLE) {
+        return false;
+    }
+
+    const auto& kernel_params = dynamic_cast<const grid_sample_params&>(params);
+    if (kernel_params.inputs.size() != 2) {
```

Comment:
The two inputs should have 4-dim tensor. Why don't you check this at this point?

Reply:
resolved
add supported layout at GetSupportedKey